packages/rs-dpp/src/identity/state_transition/identity_credit_withdrawal_transition/validation/state/validate_identity_credit_withdrawal_transition_state.rs

```diff
@@ -66,6 +66,16 @@ where
             return Ok(result);
         }
 
+        // Check revision
+        if existing_identity.get_revision() != (state_transition.get_revision() - 1) {
+            result.add_error(StateError::InvalidIdentityRevisionError {
+                identity_id: existing_identity.get_id().to_owned(),
+                current_revision: existing_identity.get_revision(),
+            });
+
+            return Ok(result);
+        }
+
```

Comment:
We also need to validate that the withdrawal id hasn't already been used.

Reply:
It's not a user's fault if any collision happens. Shall we handle it in the apply function so that if it's already exist we can change it deterministically until we find a unique one?